src/GovUk.Education.ExploreEducationStatistics.Content.Api.Tests/Controllers/DataSetFilesControllerRanksTests.cs

```diff
@@ -0,0 +1,353 @@
+using GovUk.Education.ExploreEducationStatistics.Common.IntegrationTests.WebApp;
+using GovUk.Education.ExploreEducationStatistics.Common.Model;
+using GovUk.Education.ExploreEducationStatistics.Common.Model.Data;
+using GovUk.Education.ExploreEducationStatistics.Common.Tests.Extensions;
+using GovUk.Education.ExploreEducationStatistics.Common.Tests.Fixtures;
+using GovUk.Education.ExploreEducationStatistics.Common.ViewModels;
+using GovUk.Education.ExploreEducationStatistics.Content.Api.Tests.Fixtures.Optimised;
+using GovUk.Education.ExploreEducationStatistics.Content.Model;
+using GovUk.Education.ExploreEducationStatistics.Content.Model.Database;
+using GovUk.Education.ExploreEducationStatistics.Content.Model.Extensions;
+using GovUk.Education.ExploreEducationStatistics.Content.Model.Tests.Fixtures;
+using GovUk.Education.ExploreEducationStatistics.Content.Requests;
+using GovUk.Education.ExploreEducationStatistics.Content.ViewModels;
+using Microsoft.AspNetCore.WebUtilities;
+using MockQueryable.Moq;
+using Moq;
+using Xunit;
+using ReleaseVersion = GovUk.Education.ExploreEducationStatistics.Content.Model.ReleaseVersion;
+
+#pragma warning disable CS9107 // Parameter is captured into the state of the enclosing type and its value is also passed to the base constructor. The value might be captured by the base class as well.
+
+namespace GovUk.Education.ExploreEducationStatistics.Content.Api.Tests.Controllers;
+
+// ReSharper disable once ClassNeverInstantiated.Global
+public class DataSetFilesControllerRanksTestsFixture()
```

Comment:
Cool 🙂 . I was thinking for a sec, that we'd be losing these tests. The `DataSetFileService` was written at a time when our approach changed to go full-on integration testing without unit testing services, so without this there'd be no other coverage of search ranking.

`DataSetFilesControllerRanksTests` and `ReleaseFileControllerStreamTests` will be good reference examples of how to swap out services with mocks.

Reply:
Nice one, thanks!  I'll use them as examples when I give an overview :)